build/gen_js_enums.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python
+#
+# Copyright 2015 Google Inc. All Rights Reserved.
+
+import inspect
+import optparse
+import os
+import sys
+
+USAGE = """%prog src_path dst_path
+Generate analytics enums for use in Javascript.
+
+src_path     Path to the source code root directory.
+dst_path    Path to the root directory to push/deploy GAE from."""
+
+
+def printEnums(fp, enum, indent=2):
+  indentStr = ' ' * indent
+  subEnums = []
+  for k, v in enum.__dict__.iteritems():
+    if k.startswith('_') or k.upper() != k:
+      continue
+
+    if inspect.isclass(v):
+      subEnums.append((k, v))
+      continue
+
+    fp.write("{0}{1}: ".format(indentStr, k))
+    if isinstance(v, int) or isinstance(v, long):
+      fp.write("{0}".format(v))
+    else:
+      fp.write("'{0}'".format(v))
+
+    fp.write(",\n")
+
+  for k, v in subEnums:
```

Comment:
The `RequestField` enum has embedded fields. But this generation wasn't working since I made this commit from the original patch I had. Now it generates the subEnums properly.